tests/unit/s2n_certificate_test.c

```diff
@@ -100,9 +105,13 @@ int main(int argc, char **argv)
 
     const uint8_t ocsp_data[] = "ocsp data";
 
-    struct s2n_cert_chain_and_key *chain_and_key = NULL;
-    EXPECT_SUCCESS(
-        s2n_test_cert_chain_and_key_new(&chain_and_key, S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key = NULL, s2n_cert_chain_and_key_ptr_free);
```

Comment:
dont believe this is used. can you add more than one cert and test with that: `s2n_config_add_cert_chain_and_key_to_store(... chain_and_key)`

Reply:
Other tests in this file use the default chain. A connection can only have one chain associated with it at a time.

But a chain can contain multiple certs, and it looks like all our valid test chains (that have associated private keys) only contain one certificate. I tested locally with a multi-cert chain and I'll follow up this PR with a new, multi-cert test chain.